File: Core/Models/WordHint.cs
namespace OhMyWord.Core.Models;

public class WordHint
{
    private readonly Word word;

    public int Length => word.Id.Length;
    public string Definition => word.Definition;
    public List<LetterHint> Letters { get; } = new();

    public WordHint(Word word)
    {
        this.word = word;
    }
}

File: Client/src/app/models/word-hint.ts
import { LetterHint } from './letter-hint';
import { WordHintResponse } from './responses/word-hint.response';

export class WordHint {
  readonly length: number;
  readonly definition: string;
  readonly letterHints: string[];

  constructor(init?: Partial<WordHintResponse>) {
    this.length = init?.length ?? 0;
    this.definition = init?.definition ?? '';
    this.letterHints = new Array(this.length).fill('');

    // if (init?.letters) {
    //   for (const letter of init.letters) {
    //     this.letterHints.push(new LetterHint(letter.position, letter.value));
    //   }
    // }
  }

  public static default = new WordHint();
}
